core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -42,7 +42,7 @@ import org.apache.spark.internal.Logging
 private[spark] class HadoopDelegationTokenManager(
     sparkConf: SparkConf,
     hadoopConf: Configuration,
-    fileSystems: Set[FileSystem])
+    fileSystems: () => Set[FileSystem])
```

Comment:
And the difference of `HadoopFSDelegationTokenProvider` before and after is whether to get FS in `keytabLoggedInUGI` or not.